tests/components/tuya/test_binary_sensor.py

```diff
@@ -50,12 +50,12 @@ def mock_config_entry() -> MockConfigEntry:
 
 
 @pytest.fixture
-def mock_fault_device() -> CustomerDevice:
+def mock_device_with_fault() -> CustomerDevice:
```

Comment:
It would be great if this was as close as possible to a "real" device, including valid id, name, product name, etc...
I would even recommend to add this to `conftest.py` so that in a follow-up PR it can be used in other platforms.

For example, if you have a washing machine with sensors, switches and binary sensors you can re-use the same fixture accross all platform tests.

The same applies to `mock_device_without_fault`. If you have a device with faults and a device without faults you create the two fixtures as "mock_device_xyz" and "mock_device_abc"

Reply:
Sounds like a good idea - I'll add that 😄

Reply:
If you're extra motivated, add that in a separate (preliminary) PR (no code changes - just test changes).
Then this PR becomes just about adding the sensors with updated snapshots (only code changes with snapshots - no test changes)

Reply:
> I've added details from a diagnostics dump of my dehumidifier as a mock device to conftest, and I've also added another "real" device without faults for test coverage, based on a diagnostics dump found in the issues section (from issue #108301).

Thanks for that - I've extracted those into a separate PR #148034

If you (or another user) has extra device specs we can add them to that PR, or in a follow-up.

Reply:
Thanks! I'm not really swimming in free time at the moment, so it's great that you are helping out 😇
I guess I'll wait till your PR is merged, and then fix up my branch 😊